src/main/java/cz/cvut/kbss/termit/service/repository/ResourceRepositoryService.java

```diff
@@ -169,4 +169,31 @@ private void removeFromParentDocumentIfFile(Resource instance) {
     public long getLastModified() {
         return resourceDao.getLastModified();
     }
+
+    /**
+     * Moves the triples of the document from the original vocabulary to the default context and from the
+     * new vocabulary to the context of this vocabulary.
+     *
+     * @param vOriginal original version of the vocabulary (before update)
+     * @param vNew new version of the vocabulary (after update)
+     */
+    public void rewireDocumentsOnVocabularyUpdate(final Vocabulary vOriginal, final Vocabulary vNew) {
```

Comment:
This seems quite inefficient to me. What if the document is the same? Then it will be moved around pointlessly.

Reply:
Good point, thanks for noting. Will fix this case.